### PR TITLE
fix: type safety in Matrix Dart SDK database

### PR DESCRIPTION
**Pull request by TheOneWithTheBraid**
- removes unsafe casts
- builds list using List<T>.from constructor
- removes unwanted late statement

Got the following premium bug report sent via matrix : 

bug-report

### CODE REVIEW

**Comment by TheOneWithTheBraid** [comment]
Fixes: https://github.com/krille-chan/fluffychat/issues/794

File: lib/src/database/matrix_sdk_database.dart [hunk 1]
```diff
@@ -433,7 +434,7 @@ class MatrixSdkDatabase extends DatabaseApi {
                 ? timelineEventIds.getRange(start, end).toList()
                 : []);
 
-        return await _getEventsByIds(eventIds.cast<String>(), room);
+        return await _getEventsByIds(eventIds, room);
```
Comment (krille-chan) [inline lib/src/database/matrix_sdk_database.dart:437]: This has already been fixed here: https://github.com/famedly/matrix-dart-sdk/commit/2afd7040a7fa7f8a65cab6196c0f56407334dc65

File: lib/src/database/matrix_sdk_database.dart [hunk 2]
```diff
@@ -693,14 +694,10 @@ class MatrixSdkDatabase extends DatabaseApi {
...
-                  .toList()
-                  .cast<Map<String, dynamic>>(),
-              crossSigningEntries
-                  .where((c) => c != null)
-                  .toList()
-                  .cast<Map<String, dynamic>>(),
+              List<Map<String, dynamic>>.from(
+                  childEntries.whereType<Map<String, dynamic>>()),
+              List<Map<String, dynamic>>.from(
+                  crossSigningEntries.whereType<Map<String, dynamic>>()),
```
Comment (krille-chan) [inline lib/src/database/matrix_sdk_database.dart:700]: `List<T>.from()` is not better than `list.cast<T>()` as it crashes in the same way when there is an entry which is not `T`. The difference is that `List<T>.from()` create a non-deep copy. Do we need this here? Also this is unrelated to the bug described in the commit message

File: lib/src/database/matrix_sdk_database.dart [hunk 3]
```diff
@@ -1489,7 +1486,7 @@ class MatrixSdkDatabase extends DatabaseApi {
     try {
       await clear();
       await open();
-      final json = Map.from(jsonDecode(export)).cast<String, Map>();
+      final json = Map<String, Map>.from(jsonDecode(export));
```
Comment (krille-chan) [inline lib/src/database/matrix_sdk_database.dart:1489]: Same here. This does not prevent type errors or makes anything more type safe. It just creates a non-deep copy
